pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -907,7 +1099,22 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 			activated = true
 		}
 	}
+
 	p.moveRequestCycle = p.schedulingCycle
+
+	// (no need to check the feature gate because there is always no p.inFlightPods when the feature is disabled.)
```

Comment:
in general, we shouldn't make these simplifications. We should assume that any part of the logic is buggy and check the feature gate wherever necessary.